CMSgov/mint-app
# [Easi 4365]/restricted audit translations

# EASI-4365
## Changes and Description

- Add the idea of a restricted audit
- Restrict Documents and Links based on confidentiality
- Translated Audit collection query now checks permissions when it fetches audits



## How to test this change
1. Make a model plan
2. Make a solution
3. Link a new confidential document
4. In a separate tab, log in as a non assessment, non-collaborator user.
5. In change history, assert that you cant see the confidential audits
6. Assert that you can see them for the original account

testing deleted links
1. delete the link
2. Note that an assessment user will see everything as normal
3. Look at the non-collaborator account view
4. Notice that the link is present, but all document audits are deleted (this is because the document is not queryable, so unable to mark that link as confidential)



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
